src/Parser/MarksmanshipHunter/Modules/Items/UllrsFeatherSnowshoes.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+
+import { formatNumber } from "common/format";
+import SpellLink from "common/SpellLink";
+
+const COOLDOWN_REDUCTION_MS = 800;
+
+const AFFECTED_ABILITIES = [
+  SPELLS.AIMED_SHOT.id,
+  SPELLS.MARKED_SHOT.id,
+  SPELLS.ARCANE_SHOT.id,
+  SPELLS.MULTISHOT.id,
+];
+
+class UllrsFeatherSnowshoes extends Module {
+  static dependencies = {
+    combatants: Combatants,
+    spellUsable: SpellUsable,
+  };
+
+  effectiveTrueshotReductionMs = 0;
+  wastedTrueshotReductionMs = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasFeet(ITEMS.ULLRS_FEATHER_SNOWSHOES.id);
+  }
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (AFFECTED_ABILITIES.every(id => spellId !== id)) {
+      return;
+    }
+    const trueshotIsOnCooldown = this.spellUsable.isOnCooldown(SPELLS.TRUESHOT.id);
+    if (trueshotIsOnCooldown) {
+      const reductionMs = this.spellUsable.reduceCooldown(SPELLS.TRUESHOT.id, COOLDOWN_REDUCTION_MS);
+      this.effectiveTrueshotReductionMs += reductionMs;
+    } else {
+      this.wastedTrueshotReductionMs += COOLDOWN_REDUCTION_MS;
+    }
+  }
+
+  item() {
+    return {
+      item: ITEMS.ULLRS_FEATHER_SNOWSHOES,
+      result: (
+        <dfn data-tip={`You wasted ${formatNumber(this.wastedTrueshotReductionMs / 1000)} seconds of CDR.<br/> `}>
+          You effectively reduced <SpellLink id={SPELLS.TRUESHOT.id} /> CD reduced by {formatNumber(this.effectiveTrueshotReductionMs / 1000)} in total.
```

Comment:
Little more wording here - the "reduced" after CD is redundant ("You effectively reduced Trueshot CD **reduced** by ...")

Reply:
oh, doing things in between pulls not ideal